### ISSUE TITLE
Games downloaded in 3 not recognised as already downloaded in 4

### ISSUE THREAD

**Issue by djay** [zach-morris/plugin.program.iagl#319]
I've set up the download folder the same. So not sure why it attempts to download the game again.

**Comment by zach-morris** [comment]
You got an example and/or debug log?
File matching is as follows (code [here](https://github.com/zach-morris/plugin.program.iagl/blob/v4/resources/lib/download.py#L309)):

1) Get a list of all files in the download path for this game list
2) See if there are any exact matches for a 'launch file' (certain game lists require a conf file or cmd file, etc to launch)
3) See if there are any exact matches for the file requested to be downloaded ("Mario (USA).zip" for example would match with "https://archive.org....Mario%20(USA).zip")
4) See if there are any stem matches for the file requested to be downloaded  ("Mario (USA).nes" for example would match with "https://archive.org....Mario%20(USA).zip")
5) See if the string of the file stem requested is in the requested download path (this catches any files that aren't named exactly as needed after extraction, this seems to happen mainly with hacks and translations ("Mario (USA) (T-En).nes" for example would match with "https://archive.org....Mario%20(USA) (T-En) [translation by XYZ].zip")

In order, just use the first match found, if any.  If no match, continue with download.

**Comment by djay** [comment]
here is one example.

In 4 - Sony\ PlayStation\ Portable/Daxter\ (US).iso
In 3 - Playstation_Portable/Daxter\ \(USA\)\ \(En,Fr,De,Es,It\).iso

**Comment by zach-morris** [comment]
Yeah, the archive is not the same.  The game filename is different and they don't match.
